scripts/setup-vagrant.sh

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+
+TARGET=$1
+USER=$2
+PASSWD=$3
+INSTALL=$4
+VERIFY=$5
+
+case $TARGET in
+  "el7")
+    DCTARGET=centos7
+    INSTALL_CMD="yum";;
+  "trusty"|"xenial")
+    DCTARGET=$TARGET
+    INSTALL_CMD="apt-get";;
+  *)
+    echo "[Error] Unknown target $TARGET"
+    exit 1;;
+esac
+
+echo "[Install] dependencies"
+sudo $INSTALL_CMD update
+if [[ $TARGET != 'el7' ]]; then
+  sudo apt-get -y autoremove
+fi
+
+sudo $INSTALL_CMD install -y linux-image-extra-$(uname -r)
+sudo $INSTALL_CMD install -y git curl wget
+
+# Install docker-compose
+DC="/usr/local/bin/docker-compose"
+URL="https://github.com/docker/compose/releases/download/1.9.0/docker-compose-`uname -s`-`uname -m`"
+if [[ ! -x $DC ]]; then
+  echo "[Install] docker-compose $TARGET"
+  sudo sh -c "curl -sL $URL > $DC"
+  sudo chmod +x $DC
+fi
+
+# Using docker-compose, 1) build packages, and 2) run quick tests
```

Comment:
We've to make sure we supporting building an arbitrary branch of st2. Without that, this would not be so useful. I don't see a way to specify the branch.